assets/src/domain/eventEditor/ui/tickets/hooks/useRecalculateBasePrice.ts

```diff
@@ -0,0 +1,29 @@
+import { useCallback } from 'react';
+
+import { EntityId } from '@dataServices/types';
+import { useInitialState } from '@edtrUI/tickets/ticketPriceCalculator/data';
+import { calculateBasePrice } from '@edtrUI/tickets/ticketPriceCalculator/utils';
+import { isBasePrice } from '@sharedEntities/priceTypes/predicates';
+import { usePriceMutator } from '@edtrServices/apollo/mutations';
+
+const useRecalculateBasePrice = (ticketId: EntityId): VoidFunction => {
+	// This will give us the exact state expected by `calculateBasePrice()`
+	const getDataState = useInitialState({ ticketId });
+	const { updateEntity } = usePriceMutator();
+
+	return useCallback(() => {
+		// get the list of updated prices with the amount of base price updated
+		const newPrices = calculateBasePrice(getDataState(null));
```

Comment:
`useInitialState` callback receives `initialState` and then it got destructured, it's a bit confusing
![image](https://user-images.githubusercontent.com/1477453/81731827-2da5c580-9498-11ea-8e22-4f8a53cfab95.png)

Reply:
`null` and `undefined` values are no destructured.

Reply:
makes sense, that null just caught my attention and I was thinking if it's a good idea to just call `getDataState` w/o any args and then do the check inside of `useInitialState`, smth like:
```
return useCallback<StateInitializer>(
		(initialState) => {
			return { ...(initialState && { ...initialState }), ticket, prices };
		},
		[ticket, prices]
	);
```
but I don't like very much this approach :point_up: 
so, just wrote my thoughts outloud